src/app/shared/task-provider/task-provider.component.html

```diff
@@ -1,9 +1,11 @@
 <div [formGroup]="taskProviderFormGroup" *ngIf="isInitialized">
     <div class="layout-row">
         <mat-form-field floatLabel="always" class="task-provider-select flex">
+            <mat-icon matSuffix *ngIf="!isUpdateMode">edit</mat-icon>
+            <mat-icon matSuffix *ngIf="isUpdateMode">block</mat-icon>
             <mat-label></mat-label>
             <mat-select [formControlName]="selectedTaskProviderKey" placeholder="Select a Task Provider"
-                        (selectionChange)="onProviderChange($event.value)">
+                        (selectionChange)="onProviderChange($event.value)" [disabled]="isUpdateMode">
```

Comment:
Can't we set the disabled property in the component, while creating the formcontrol?

Reply:
Done.